### PR TITLE
Fix moving process element bug

### LINKED ISSUE #1213: 🐛 [BUG] - Moving process element to change order fails
### Summary

Moving single elements to change order does not seem to work anymore.

Maybe a change in https://github.com/dlr-gtlab/gtlab-core/pull/1168 lead to this problem

### Expected Behaviour

If two process elements are on the same hierarchy I would like to change their order by drag and drop. This currently does not work

### Reproduction steps

```bash
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error
```

### Screenshots

grafik  

Switching these calculator elements is not possible

### Logs

```bash
[15:13:46]  compared object mementos are identical!
```

### GTlab environment

```bash
GTlab Version: Core master build
```

### OS

Windows

### PR DESCRIPTION
**Pull request by jensschmeink**
## Description

Closes #1213 

## Checklist:

- [ ] A test for the new functionality was added (if applicable).
- [ ] All tests run without failure.
- [ ] The changelog has been extended, if this MR contains important changes from the users's point of view.
- [ ] The new code complies with the GTlab's style guide.
- [ ] New interface methods / functions are exported via `EXPORT`. Non-interface functions are NOT exported.
- [ ] The number of code quality warnings is not increasing.

### CODE REVIEW

**Comment by rainman110** [comment]
@jensschmeink Please rebase to the current master to enable cppcheck

**Review by rainman110 — CHANGES_REQUESTED** [review]
The move elements function is now too long. Can you please subdivide it into smaller sub-functions. It is currently very hard to read and understand.

**Comment by jensschmeink** [comment]
> The move elements function is now too long. Can you please subdivide it into smaller sub-functions. It is currently very hard to read and understand.

I divided parts of the function in seperate functions

**Comment by jensschmeink** [comment]
@rainman110 after the warning is hidden now, can we merge?

**Review by rainman110 — CHANGES_REQUESTED** [review]
So after having functional tests, this is my review on the code.

Please just add docstrings to the new functions in the header, see comments.

File: src/app/dock_widgets/process/gt_processdock.h [hunk 1]
```diff
@@ -310,6 +310,20 @@ public slots:
...
 
+    /**
+     * @brief findObjectsOfIdenticalParentByMapping
+     * @param source - indices from the view
+     * @return list of objects which corresponds to the indices including a
+     * check if all elements have the same parent
+     *
+     * Teh objects are in the order of the row indices of the source elements
+     */
+    QList<GtObject*> findObjectsOfIdenticalParentByMapping(
```
[the remark this answers is not in the document]
Comment (jensschmeink) [reply src/app/dock_widgets/process/gt_processdock.h:324]: I needed to add a const version of the basic funtion "mapToSource()" which I added now.

File: src/app/dock_widgets/process/gt_processdock.h [hunk 2]
```diff
@@ -310,6 +311,33 @@ public slots:
...
+    /**
+     * @brief findObjectsOfIdenticalParentByMapping
+     * For the indices the objects from the datamodel are searched.
+     * The indices are sorted based on their row number for this.
+     * Additionally a check is performaned if all found objects have
+     * the same parent
+     * @param source - indices from the view
+     * @return list of objects which corresponds to the indices including a
+     * check if all elements have the same parent
+     */
```
Comment (rainman110) [inline src/app/dock_widgets/process/gt_processdock.h:323]:
```suggestion
     * @brief Returns the objects from the model given their model index.
     *
     * The indices are sorted based on their row number for this.
     *
     * *Note*: A check is performed to make sure that all objects have
     * the same parent. If not the case, an empty list is returned.
     *
     * @param source - Model indices
     * @return list of objects which corresponds to the indices
     */
```

File: src/app/dock_widgets/process/gt_processdock.h [hunk 3]
```diff
@@ -310,6 +311,33 @@ public slots:
...
+    QList<GtObject*> findObjectsOfIdenticalParentByMapping(
```
Comment (rainman110) [inline src/app/dock_widgets/process/gt_processdock.h:325]:
```suggestion
    QList<GtObject*> findObjectsOfSameParent(
```

File: src/app/dock_widgets/process/gt_processdock.h [hunk 4]
```diff
@@ -310,6 +311,33 @@ public slots:
...
+            const QList<QModelIndex>& source) const;
+
+    /**
+     * @brief moveConnections
+     * After the move of the objects given by the finalyMovedObjects from
```
Comment (rainman110) [inline src/app/dock_widgets/process/gt_processdock.h:330]: I still dont understand, what the function is doing. The sentence is very long. Please write @brief a short description and optionally add details, if necessary after a blank line.

Comment (jensschmeink) [reply src/app/dock_widgets/process/gt_processdock.h]: @rainman110 I added some more lines to describe the functino

**Review by rainman110 — APPROVED** [review]

**Pull request merged**